bin/main.rb

```diff
@@ -1,2 +1,62 @@
 #!/usr/bin/env ruby
-puts "Hello World"
+
+require_relative '../lib/gamelogic.rb'
+
+def display_invalid_choice 
+    
+  puts "\n\nINVALID CHOICE\n\n You can choose only EMPTY cells using letters & and numbers: a1, a2, b3, c4, etc.\n\n"
+  sleep 2.0
+  puts display_board
+
+end
+
+first_game = Game.new
+board = Board.new
```

Comment:
From an OOP point of view, this is not good.
The Board is an integral part of the game. It cannot exists outside the game, it cannot exist if the game does not exist.

From that statement, If you instantiate the Game, the game comes to the world with the Board (and the players) and if you destroy the Game, it dies after killing himself the Board (and the players)

That is the **OOP**. All that talk to tell you guys to have the board as an attribute of the game.
```ruby
class Game
   attr_reader :board 
   ....
end
```

Reply:
thank, I think I get the idea now. but I'm not sure how to fix it. could you give me any advice or a link to documentation or tutorial